### DISCUSSION TITLE
Add Setting for Recent-Messages Provider

### DISCUSSION THREAD

**Discussion by Nerixyz** [Chatterino/chatterino2#4299] [Ideas]
There are several Recent-Messages hosts nowadays and people might prefer one over the other or use one in the case their preferred instance is unreachable. This option already exists but only through environment variables. This isn't end-user friendly, thus a setting (e.g. a text box) should be added.

In the future, the chatterino API could suggest an active instance in the case that the current one is unreachable.

**Comment by Mm2PL** [comment]
> In the future, the chatterino API could suggest an active instance in the case that the current one is unreachable.

We probably should have it in a similar way to image uploader services if anything. Having it being highly automated doesn't sit well to me for some reason. I like knowing and picking the websites or services Chatterino contacts and I feel others probably do as well

**Reply by Nerixyz** [reply]
I agree, it seems like a privacy risk. Maybe it could send a message like `Message History unavailable (5xx) - do you want to switch the message history host to <host>?` or similar.

**Reply by LosFarmosCTL** [reply]
In general, a list for several providers could be a nice option, with chatterino switching to the next one when the previous one is unavailable.

There are already some people that simulate something like that with their own middleware API, but a built-in solution without environment variables would be nice.

**Reply by kornes** [reply]
as 3rd party service it really makes sense to move it to External tools, also like LosFarmosCTL idea with editable ordered list and automatic fallback

**Reply by Felanbird** [reply]
I'm fine with adding an input box to change the `env`, but any other implementation means we are forcing an EXTREME load onto another hoster, and I'd assume only Leppunen's server would be in the position to handle the load during downtime.

I spoke to randers the first day and he mentioned making a little 😆 which caused the first days issue, tho I'm unsure if said xd is the cause of the issues in the following 2 days.

**Reply by leon-richardt** [reply]
I agree with @Felanbird, automatically switching to another recent-messages instance when the default instance goes down would essentially trigger a DDoS attack once all clients switch to the next one in the list. This might well be too much to handle for smaller instances that were only meant for private use.

**Reply by LosFarmosCTL** [reply]
Just to clarify, I meant that as a user-configurable list, without any preset defaults as fallbacks. So I wouldn't expect a lot of users to actually use that option, the subset of people even knowing what recent-messages is, is already not that big.

**Reply by pajlada** [reply]
To start, we shouldn't worry about fallbacks.
env variable > setting field, that should be simple enough for users & simple enough implementation-wise
Adding a user-configurable list, even if we only expect very very few users to use it, means we complicate the implementation for every user
